lib/api/filter.rb

```diff
@@ -138,16 +132,10 @@ def single_expression(field, operator, value)
       {operator => {"field" => field, "value" => value}}
     end
 
-    def target_class(klass, reflections)
-      if reflections.empty?
-        klass
-      else
-        target_class(klass.reflections_with_virtual[reflections.first.to_sym].klass, reflections[1..-1])
-      end
-    end
-
-    def virtual_or_physical_attribute?(klass, attribute)
-      klass.attribute_method?(attribute) || klass.virtual_attribute?(attribute)
+    def split_filter_string(str, operator)
+      filter_attr, _op, filter_value = str.partition(operator)
+      *associations, attr_name = filter_attr.strip.split(".")
```

Comment:
I'm not sure if it's possible, but are there cases where the filter_attr will _not_ have a `.`?  The reason I ask is

```ruby
*associations, attr_name = "x".split(".")
associations # => []
attr_name    # => "x"
```

when compared to

```ruby
*associations, attr_name = "x.y".split(".")
associations # => ["x"]
attr_name    # => "y"
```

That is, the trailing object is put in attr_name, regardless of the preceding associations, and I wasn't sure if the trailing thing was _always_ an actual attr_name. (Hope that made sense)

Reply:
This code got moved, but the question still stands.

Reply:
Yes, good question. Most of the time the `filter_attr` will not have a `.`.
This is the base case and most common case for our tests: e.g.: https://github.com/ManageIQ/manageiq-api/blob/column_type/spec/lib/api/filter_spec.rb#L3

```
# model = "Vm"
str = "name"
associations, attr_name = [], "name

str = "host.name" 
associations, attr_name = ["host"], "name"
```

Reply:
Ok perfect then this code is good